### ISSUE TITLE
CollectionView doesn't animate moves properly.

### ISSUE THREAD

**Issue by aranasaurus** [danthorpe/TaylorSource#48]
`processItemChanges()` is deleting and then inserting which causes a kind of teleportation effect. `UICollectionView` has `moveItemAtIndexPath(toIndexPath:)` for moving things (`UITableView` does not). It'd be nice if TaylorSource used that.

**Comment by danthorpe** [comment]
:+1: 

Out of curiosity, I was going to use [Dwifft](https://github.com/jflinter/Dwifft) - thoughts?

**Comment by aranasaurus** [comment]
Interesting, I'd not seen that before. I thought that was the point of `performBatchUpdates` though?

**Comment by danthorpe** [comment]
Well, yes I think that is the point of Apple's perform batch updates - but it's pretty tricky to assimilate change sets into a suitable stream for it, which is what TayorSource is trying to do with `processItemChanges()`

There have been a couple of attempts to do this more robustly, I'm just thinking that it's not really the core value proposition of TaylorSource, and it's easily the biggest pain point. So, it makes sense to offload this to a specialist tool.

**Comment by aranasaurus** [comment]
oh, right... I forget that TaylorSource is used _without_ YapDatabase, which makes those methods super simple since it generates those changesets, right? That PR I put up also only updates the animations in the YapDBFactory UICollectionView extension, so there's probably another one somewhere that also needs to be updated with the same.

**Comment by aranasaurus** [comment]
I think it makes perfect sense to offload that to a specialist tool.

**Comment by danthorpe** [comment]
Yeah, and I think for using it with YapDatabase, because the database sends change notifications, I think it'll be "easy" to massage that into a structure which Dwifft can work with.

So the datasource (Basic / NSFRC / YapDB) -> [Changeset] -> Dwiftt -> View

Lots of handwaving here though....

**Comment by aranasaurus** [comment]
A quick look at the Dwifft+UIKit stuff shows that it's going to have the same result as I brought up here though. It is using delete/insert and not calling move at all :(

**Comment by danthorpe** [comment]
Oh.

Well, it's open source right? :)

**Comment by aranasaurus** [comment]
Indeed. :)

**Comment by aranasaurus** [comment]
https://github.com/jflinter/Dwifft/issues/1 haha, it was issue number 1... and he shot it down.

**Comment by aranasaurus** [comment]
oh, and I missed it but `UITableView` _does_ have an equivalent... it's just called `moveRowAtIndexPath:toIndexPath`.

There are also Section equivalents, which it looks like if YapDatabase will generate a .Move type of SectionChange, TaylorSource will just ignore it. So maybe there's a fair few more updates needed to this bit than I thought.
